src/components/dbManager.js

```diff
@@ -1,234 +1,304 @@
+        const couch = Db.getCouch();
+        const db = couch.db.use(databaseName);
+
+        // Do a sanity check to confirm the username is an admin of the database
+        const perms = await couch.request({db: databaseName, method: 'get', path: '/_security'})
+        const usernameIsAdmin = perms.admins.names.includes(username)
+
+        if (!usernameIsAdmin) {
+            return false
+        }
+
+        try {
+            await this.configurePermissions(db, username, applicationName, options.permissions);
+        } catch (err) {
+            //console.log("update database error");
+            //console.log(err);
+        }
+
+        return true;
     }
 
-    // Create validation document so only owner users in the write list can write to the database
-    let writeUsersJson = JSON.stringify(writeUsers);
-    let deleteUsersJson = JSON.stringify(deleteUsers);
-
-    try {
-      const writeFunction =
-        '\n    function(newDoc, oldDoc, userCtx, secObj) {\n        if (' +
-        writeUsersJson +
-        ".indexOf(userCtx.name) == -1) throw({ unauthorized: 'User is not permitted to write to database' });\n}";
-      const writeDoc = {
-        validate_doc_update: writeFunction,
-      };
-
-      await this._insertOrUpdate(db, writeDoc, '_design/only_permit_write_users');
-    } catch (err) {
-      // CouchDB throws a document update conflict without any obvious reason
-      if (err.reason != 'Document update conflict.') {
-        throw err;
-      }
+    async deleteDatabase(databaseName, username) {
+        const couch = Db.getCouch();
+
+        // Do a sanity check to confirm the username is an admin of the database
+        const perms = await couch.request({db: databaseName, method: 'get', path: '/_security'})
+        const usernameIsAdmin = perms.admins.names.includes(username)
+
+        if (!usernameIsAdmin) {
+            return false
+        }
+
+        // Create database
+        try {
+            return await couch.db.destroy(databaseName);
+        } catch (err) {
+            // The database may already exist, or may have been deleted so a file
+            // already exists.
+            // In that case, ignore the error and continue
+            //console.log(err);
+        }
     }
 
-    if (permissions.write == 'public') {
-      // If the public has write permissions, disable public from deleting records
-      try {
-        const deleteFunction =
-          '\n    function(newDoc, oldDoc, userCtx, secObj) {\n        if (' +
-          deleteUsersJson +
-          ".indexOf(userCtx.name) == -1 && newDoc._deleted) throw({ unauthorized: 'User is not permitted to delete from database' });\n}";
-        const deleteDoc = {
-          validate_doc_update: deleteFunction,
+    async configurePermissions(db, username, applicationName, permissions) {
+        permissions = permissions ? permissions : {};
+
+        let owner = username;
```

Comment:
`username` is not used further in the function, could call it `owner` right away or add jsdoc that it's owner's username